domains/pallets/messenger/src/lib.rs

```diff
@@ -1247,6 +1252,19 @@ where
 
 impl<T: Config> sp_domains::DomainBundleSubmitted for Pallet<T> {
     fn domain_bundle_submitted(domain_id: DomainId) {
-        DomainChainAllowlistUpdate::<T>::remove(domain_id);
+        // NOTE: clear the updates leave an empty value but does not delete the value for the
+        // domain completely because in the invalid extrinsic root fraud proof the prover need
+        // to generate a proof-of-empty-value for the domain.
+        DomainChainAllowlistUpdate::<T>::mutate(domain_id, |maybe_updates| {
+            if let Some(ref mut updates) = maybe_updates {
+                updates.clear();
+            }
+        });
+    }
+}
+
+impl<T: Config> sp_domains::OnDomainInstantiated for Pallet<T> {
+    fn on_domain_instantiated(domain_id: DomainId) {
```

Comment:
`ValueQuery` or `OptionalQuery` just controls what to return if the requested key-value does not exist in the state, which doesn't mean it actually stores a `Default::default` or `Option` value in the state.

When the key-value does not exist in the state `read_proof` will return a `MissingValue` error which makes sense because the storage proof is a proof-of-existence, not a proof-of-absence. What this callback does is manually store an empty value for the domain so we can generate a storage proof to prove this is an empty value.